### PR TITLE
docs(self-managed): mention the change of using image.tag

### PR DESCRIPTION
**Pull request by aabouzaid**
## Description

Task: https://github.com/camunda/camunda-platform-helm/issues/2067

## When should this change go live?

- [x] This is a bug fix, security concern, or something that needs **urgent release support**.
- [ ] This is already available but undocumented and should be released within a week.
- [ ] This on a **specific schedule** and the assignee will coordinate a release with the DevEx team. (apply `hold` label or convert to draft PR)
- [ ] This is part of a scheduled alpha or minor. (apply alpha or minor label)
- [ ] There is **no urgency** with this change and can be released at any time.

## PR Checklist

- [x] My changes are for an **already released minor** and are in `/versioned_docs` directory.
- [x] My changes are for the **next minor** and are in `/docs` directory (aka `/next/`).

- [ ] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned the Engineering DRI or delegate.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @camunda/tech-writers as a reviewer.

### CODE REVIEW

**Review by conceptualshark — DISMISSED** [review]

File: docs/self-managed/setup/upgrade.md [hunk 1]
```diff
@@ -117,6 +117,14 @@ For a smooth upgrade, always use the same Helm CLI version corresponding with th
 
 ## From Camunda 8.4 to 8.5
 
+### Helm chart 10.2.0+
+
+As of this Helm chart version, the image tags for all components are independent, and do not reference the global image tag. The value of the key `global.image.tag` is `null`, and each component now sets its own version.
+
+With his change, Camunda applications no longer require a unified patch version. For example, a given installation may use Zeebe version 8.5.1, and Operate version 8.5.2. Note that only the patch version can differ between components.
```
Comment (akeller) [inline docs/self-managed/setup/upgrade.md:124]:
```suggestion
With this change, Camunda applications no longer require a unified patch version. For example, a given installation may use Zeebe version 8.5.1, and Operate version 8.5.2. Note that only the patch version can differ between components.
```

Just saw this typo. I did not do an exhaustive read-through of this PR, so there might be others.

**Review by akeller — CHANGES_REQUESTED** [review]

**Comment by aabouzaid** [comment]
@conceptualshark FYI, I just made a small change and added a deprecation note because that key will be actually removed as it's confusing and not needed anymore.

**Review by conceptualshark — APPROVED** [review]
Looks good on my end!

**Comment by akeller** [comment]
Merging - this will get released later today with the alpha3 changes.

**Pull request merged**